research/2022-07-15-relay-anonymity.md

```diff
@@ -0,0 +1,351 @@
+Only after completing this analysis, and after addressing possibly upcoming issues, we will declare Waku v2 as privacy-preserving with respect to metadata protection.
+
+
+### Anonymity
+
+Privacy and anonymity are closely linked.
+Both the identity of a user and data that allows inferring a user's identity should be part of the privacy policy.
+For the purpose of analysis, we want to have a clearer separation between these concepts.
+
+We define anonymity as *unlinkablity of users' identities and their shared data and/or actions*.
+
+We subdivide anonymity into *receiver anonymity* and *sender anonymity*.
+
+#### Receiver Anonymity
+
+We define receiver anonymity as *unlinkability of users' identities and the data they receive and/or related actions*.
+The data transmitted via Waku relay must be a [Waku message](https://rfc.vac.dev/spec/14/), which contains a content topic field.
+Because each message is associated with a content topic, and each receiver is interested in messages with specific content topics,
+receiver anonymity in the context of Waku corresponds to *subscriber-topic unlinkability*.
+An example for the "action" part of our receiver anonymity definition is subscribing to a specific topic.
+
+The Waku message's content topic is not related to the libp2p pubsub topic.
+For now, Waku uses a single libp2p pubsub topic, which means messages are propagated via a single mesh of peers.
+With this, the receiver discloses its participation in Waku on the gossipsub layer.
+We will leave the analysis of libp2p gossipsub to a future article within this series, and only provide a few hints and pointers here.
+
+Waku offers k-anonymity regarding content topic interest in the global adversary model.
+[K-anonymity](https://en.wikipedia.org/wiki/K-anonymity) in the context of Waku means an attacker can link receivers to content topics with a maximum certainty of $1/k$.
+The larger $k$, the less certainty the attacker gains.
+Receivers basically hide in a pool of $k$ content topics, any subset of which could be topics they subscribed to.
+The attacker does not know which of those the receiver actually subscribed to,
+and the receiver enjoys [plausible deniability](https://en.wikipedia.org/wiki/Plausible_deniability#Use_in_cryptography) regarding content topic subscription.
+Assuming there are $n$ Waku content topics, a receiver has $n$-anonymity with respect to association to a specific content topic.
+
+Technically, Waku allows distributing messages over several libp2p pubsub topics.
+This yields $k$-anonymity, assuming $k$ content topics share the same pubsub topic.
+However, if done wrongly, such sharding of pubsub topics can breach anonymity.
+A formal specification of anonymity-preserving topic sharding building on the concepts of [partitioned topics](https://specs.status.im/spec/10#partitioned-topic) is part of our roadmap.
+
+Also, Waku is not directly concerned with 1:1 communication, so for this post, 1:1 communication is out of scope.
```

Comment:
for clarity, I think it is worth mentioning that 1:1 channel can be implemented on top of pub/sub mechainsm, otherwise people might get confused

Reply:
added a sentence in 16036a3